src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -167,15 +166,11 @@ export const getTopNavConfig = (
     }
     const state = {
       input: {
-        ...vis.serialize(),
-        id: embeddableId ? embeddableId : uuid.v4(),
-      },
+        savedVis: vis.serialize(),
```

Comment:
`savedVis` won't exist in by-value mode. Why this change?

Reply:
I don't know the visualize embeddable too well, but all the changes in this file were required to make visualize work with the new stateTransfer types. I added a unified `EmbeddablePackageState` with an `input` key which means that dashboard doesn't have to worry at all about 'by reference' vs 'by value' when adding an embeddable.

So it seems like this change here is done for the by reference case.